### PR TITLE
Context

### PR DESCRIPTION
**Pull request by WyriHaximus**
Absolutely loving this middleware but needed a way to add the request UUID to the logging context. So came up with this 😎

### CODE REVIEW

**Comment by oscarotero** [comment]
I'm agree that having a way to generate a context would be nice, but it's not needed in your example, because there's `%{VARNAME}n`, to log any request attributes and `%{VARNAME}i` to log any request header. There's also a `%a` to log specifically the client ip. You can customice the log format using these patterns.

**Comment by WyriHaximus** [comment]
True but I intend to ship these logs off to loggly as well, and having a specific field with for example the UUID or clientIP makes it a lot easier to group all the messages.

File: src/AccessLog.php [hunk 1]
```diff
@@ -105,6 +110,20 @@ public function hostnameLookups($hostnameLookups = true)
     }
 
     /**
+     * Set context callable
+     *
+     * @param bool $context
+     *
+     * @return self
+     */
+    public function context(callable $context = null)
```
Comment (oscarotero) [inline src/AccessLog.php]: The type in phpdocs is wrong. And maybe makes no sense allowing null values here.

**Review by oscarotero — CHANGES_REQUESTED** [review]
I think we should have a test for this feature.

File: src/AccessLog.php (hunk 1 above)
Comment (WyriHaximus) [reply src/AccessLog.php]: Makes sense 👍

**Comment by WyriHaximus** [comment]
Added a test and fixed your other comment

**Comment by oscarotero** [comment]
👍 thanks!

**Pull request merged**